dbclient/MLFlowClient.py

```diff
@@ -11,13 +11,117 @@
+                        raise result.exception()
+
+        end = timer()
+        logging.info("Complete MLflow Runs Export Time: " + str(timedelta(seconds=end - start)))
+
+
+    '''
+    # Later we may consume the exported runs as the following
+    def import_mlflow_runs(...):
+        con = sqlite3.connect(log_sql_file, check_same_thread=False)
+        cur = con.cursor()
+
+        # Later, we might enforce start_time > xxx
+        cur.execute("SELECT * FROM runs")
+        runs = cur.fetchmany(1000)
+        while(len(runs) > 0):
+            # parallelize here
+            for run in runs:
+                run_id = run[0]
+                start_time = run[1]
+
+                run_obj = json.loads(run[2])
+                info = run_object['info']
+                metrics = run_object['metrics']
+                params = run_object['params']
+                tags = run_object['tags']
+    '''
+
+
+    def _export_runs_in_an_experiment(self, log_sql_file, experiment_str, checkpointer, error_logger):
+        experiment_id = json.loads(experiment_str).get('experiment_id')
+        logging.info("Working on runs for experiment_id: " + experiment_id)
+        # We checkpoint by experiment_id
+        if checkpointer.contains(experiment_id):
+            return
+        page_continue = True
+        token = None
+        is_there_exception = False
+        # Unlike experiments which usually don't have too much number of data,
+        # we must do page_token handling for runs.
+        while page_continue:
+            try:
+                runs = self.client.search_runs(experiment_id, run_view_type=ViewType.ACTIVE_ONLY, max_results=3000, page_token=token)
+            except RestException as error:
+                logging.info(f"search runs failed for id: {experiment_id}. Logging it to error file...")
+                error_logger.error(error.json)
+                is_there_exception = True
+            else:
+                # With multi-threading, we need to give enough timeout for each thread's connection to be made
+                # Upon DatabaseLock timeout exception, one can simply rerun the command since the checkpoint saves the
+                # progress
+                con = sqlite3.connect(self.export_dir + log_sql_file, timeout=180)
+                with con:
+                    for run in runs:
+                        self._save_run_data_to_sql(con, run)
+                con.close()
+                token = runs.token
+                page_continue = token is not None
+
+        if not is_there_exception:
+            checkpointer.write(experiment_id)
+
+    @classmethod
+    def _save_run_data_to_sql(cls, con, run):
+        run_id = run.info.run_id
+        start_time = run.info.start_time
+        run_object = {
+            "info": dict(run.info),
+            "metrics": dict(run.data.metrics),
+            "params": dict(run.data.params),
+            "tags": dict(run.data.tags)
+        }
```

Comment:
is this format needed for import ?

Reply:
Not this particular format, but we do need info, metrics, params, tags separate objects.